autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/saving/AutofillSavingCredentialsDialogFragment.kt

```diff
@@ -122,6 +134,13 @@ class AutofillSavingCredentialsDialogFragment : BottomSheetDialogFragment(), Cre
         ViewModelProvider(this, viewModelFactory)[AutofillSavingCredentialsViewModel::class.java]
     }
 
+    private val wasUsernameBackFilled: Deferred<Boolean> = lifecycleScope.async(start = LAZY) {
```

Comment:
curious why are you using this pattern of async-await, on a different coroutine, instead of just having this logic inside a suspend method that executes inside a `withContext`?

Reply:
also want to ensure the value is cached for the duration of the dialog’s life. essentially, following the approach from here: [Knowledge Sharing: Lazy initialization of a val that is expensive to execute](https://app.asana.com/0/1204153223030487/1205190818902277/f)